### PR TITLE
Implement artifactory authenticator plugin + artifactory publishing for IJ plugins

### PR DESCRIPTION
**Pull request by ZacSweers**
This implements support for uploading intellij plugins to artifactory using gradle-property-gated values and a new (extremely small) plugin for supporting authorized access to artifactory.

The new plugin is under `:skate-plugin:artifactory-authenticator` and just exposes a single setting to configure its values + hook into intellij's plugin repositories hook for authentication.

This also cleans up some common IJ plugin configs since we now have multiple.

Screenshot 2023-07-21 at 1 09 44 PM

### CODE REVIEW

File: skate-plugin/artifactory-authenticator/src/main/kotlin/com/slack/intellij/artifactory/AuthConfig.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  BoundSearchableConfigurable(
+    displayName = AuthBundle.message("artifactoryAuth.configuration.title"),
+    helpTopic = AuthBundle.message("artifactoryAuth.configuration.title"),
+    _id = "com.slack.intellij.artifactory"
+  ) {
+  private val settings =
+    ApplicationManager.getApplication().getService(AuthPluginSettings::class.java)
+
+  override fun createPanel(): DialogPanel = panel {
+    resizableColumn()
```
Comment (ZacSweers) [inline skate-plugin/artifactory-authenticator/src/main/kotlin/com/slack/intellij/artifactory/AuthConfig.kt:41]: @kateliu20 I _think_ this is the right solution for what you were trying to do here: #449

Comment (kateliu20) [reply skate-plugin/artifactory-authenticator/src/main/kotlin/com/slack/intellij/artifactory/AuthConfig.kt:41]: Thank you for this 🙏

File: build.gradle.kts [hunk 2]
```diff
@@ -205,20 +214,20 @@ subprojects {
...
   }
 
-  val isSkatePlugin = project.path == ":skate-plugin"
+  val isIntelliJPlugin = project.hasProperty("INTELLIJ_PLUGIN")
   pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
     tasks.withType<KotlinCompile>().configureEach {
       compilerOptions {
         val kotlinVersion =
-          if (isSkatePlugin) {
+          if (isIntelliJPlugin) {
```
Comment (kateliu20) [inline build.gradle.kts:222]: Does intelliJ plugin right now include both the skate plugin and artifactory-authenticator?

File: skate-plugin/artifactory-authenticator/src/main/kotlin/com/slack/intellij/artifactory/AuthPluginSettings.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+package com.slack.intellij.artifactory
+
+import com.intellij.ide.plugins.auth.PluginRepositoryAuthListener
+import com.intellij.openapi.components.BaseState
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.SimplePersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+
+@Service(Service.Level.APP)
```
Comment (kateliu20) [inline skate-plugin/artifactory-authenticator/src/main/kotlin/com/slack/intellij/artifactory/AuthPluginSettings.kt:25]: I know for the skate plugin settings we used `Service.Level.PROJECT`. is there a difference between that and APP?

File: skate-plugin/artifactory-authenticator/README.md
Comment (kateliu20) [inline skate-plugin/artifactory-authenticator/README.md:1]: Nice 👍

**Review by kateliu20 — APPROVED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (ZacSweers) [reply build.gradle.kts:222]: This is just for build logic, they're still both separate plugins

File: skate-plugin/artifactory-authenticator/src/main/kotlin/com/slack/intellij/artifactory/AuthPluginSettings.kt (hunk 3 above)
Comment (ZacSweers) [reply skate-plugin/artifactory-authenticator/src/main/kotlin/com/slack/intellij/artifactory/AuthPluginSettings.kt:25]: Yep, plug-in repositories aren't project specific so we use app scope (I.e. IDE-wide)

**Pull request merged**